include/internal/ktls.h

```diff
@@ -67,11 +68,20 @@ static ossl_inline int ktls_send_ctrl_message(int fd, unsigned char record_type,
     return -1;
 }
 
+
+static ossl_inline int ktls_read_record(int fd, void *data, size_t length)
+{
+    return -1;
+}
+
 #   else                        /* KERNEL_VERSION */
 
 #    include <netinet/tcp.h>
 #    include <linux/tls.h>
 #    include <linux/socket.h>
+#    include "openssl/ssl3.h"
+#    include "openssl/tls1.h"
+#    include "openssl/evp.h"
 
 #    ifndef SOL_TLS
 #     define SOL_TLS 282
```

Comment:
When you include <linux/socket.h> just a few lines above, and the LINUX_VERSION_CODE is already
checked why do you define SOL_TLS here?

Reply:
This is a bit silly, but we didn't export this anywhere. I'll send a patch to the kernel to fix this.

Reply:
We don't have SOL_TLS exported at the moment. I'll send a patch to the kernel, but it will take a while to propagate. This shouldn't be arch specific.

Reply:
And how about the TCP_ULP below ?

Reply:
> We don't have SOL_TLS exported at the moment. I'll send a patch to the kernel, but it will take a while to propagate. This shouldn't be arch specific.

BTW: I see this define in the kernel sources:
include/linux/socket.h:#define SOL_TLS		282

why does it not work?